### PR TITLE
feat(surveys): added a customizable submit button text color

### PR DESCRIPTION
**Pull request by dmarticus**
[Hospitable asked for this](https://posthog.slack.com/archives/C074A08LP1B/p1722508627577449), and it was super easy to do. Feels worth it.

Ships with the posthog-js changes here: https://github.com/PostHog/posthog-js/pull/1339

Video demo: https://www.loom.com/share/07e3409daded4bf4908442286a88b009?sid=a7c6d250-bb82-4ce0-905c-cc89a7d20716

### CODE REVIEW

File: frontend/src/scenes/surveys/SurveyCustomization.tsx [hunk 1]
```diff
@@ -99,17 +99,23 @@ export function Customization({ appearance, surveyQuestionItem, onAppearanceChan
...
+                        <div className="mt-2">Placeholder text</div>
                         <LemonInput
                             value={appearance?.placeholder || defaultSurveyAppearance.placeholder}
                             onChange={(placeholder) => onAppearanceChange({ ...appearance, placeholder })}
                             disabled={!surveysStylingAvailable}
                         />
                     </>
                 )}
-                <div className="mt-2">
+                <div className="mt-4">
```
Comment (dmarticus) [inline frontend/src/scenes/surveys/SurveyCustomization.tsx:118]: noticed that the customization elements seemed too crowded; added a little space here to let the different parts breathe.

Before: 
image

After: 
image

File: frontend/src/scenes/surveys/SurveyCustomization.tsx [hunk 2]
```diff
@@ -99,17 +99,23 @@ export function Customization({ appearance, surveyQuestionItem, onAppearanceChan
...
+                <div className="mt-2">Button text color</div>
+                <LemonInput
+                    value={appearance?.submitButtonTextColor}
+                    onChange={(submitButtonTextColor) => onAppearanceChange({ ...appearance, submitButtonTextColor })}
+                    disabled={!surveysStylingAvailable}
+                />
                 {surveyQuestionItem.type === SurveyQuestionType.Open && (
                     <>
-                        <div className="mt-2">Placeholder</div>
+                        <div className="mt-2">Placeholder text</div>
```
Comment (dmarticus) [inline frontend/src/scenes/surveys/SurveyCustomization.tsx:110]: imo this is clearer than just saying "Placeholder"

File: frontend/src/scenes/surveys/surveyActivityDescriber.tsx [hunk 3]
```diff
@@ -164,6 +164,7 @@ const surveyActionsMapping: Record<
             backgroundColor: 'background color',
             submitButtonColor: 'submit button color',
             submitButtonText: 'submit button text',
+            submitButtonTextColor: 'submit button text color',
```
Comment (dmarticus) [inline frontend/src/scenes/surveys/surveyActivityDescriber.tsx:167]: added to the activity log

File: frontend/src/scenes/surveys/constants.tsx [hunk 4]
```diff
@@ -32,6 +32,7 @@ export const SurveyUrlMatchTypeLabels = {
 export const defaultSurveyAppearance = {
     backgroundColor: '#eeeded',
     submitButtonColor: 'black',
+    submitButtonTextColor: 'white',
```
Comment (dmarticus) [inline frontend/src/scenes/surveys/constants.tsx:35]: default

**Review by Phanatic — APPROVED** [review]
Looks good, ship it

**Pull request merged**